src/corecel/sys/ScopedDeviceProfiling.cuda.cc

```diff
@@ -0,0 +1,64 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/sys/ScopedDeviceProfiling.cuda.cc
+//---------------------------------------------------------------------------//
+#include "ScopedDeviceProfiling.hh"
+
+#include "celeritas_config.h"
+#include "corecel/device_runtime_api.h"
+
+// Profiler API isn't included with regular CUDA API headers
+#include <cuda_profiler_api.h>
+
+#include "corecel/Assert.hh"
+#include "corecel/io/Logger.hh"
+
+#include "Device.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Activate device profiling.
+ */
+ScopedDeviceProfiling::ScopedDeviceProfiling()
+{
+    if (celeritas::device())
+    {
+        try
+        {
+            CELER_CUDA_CALL(cudaProfilerStart());
```

Comment:
Just as a note, NVIDIA [recommends](https://docs.nvidia.com/nsight-compute/ProfilingGuide/index.html#range-replay-define-range) (last bullet) to use the driver API. I'm not sure if it's really worth linking against the driver library and having a separate `CELER_CUDA_DRIVER_CALL` macro just for that case.

Reply:
> the CUDA runtime API may not trigger these if no CUDA context is active on the calling thread

Since we always initialize the context (using `cudaSetDevice` and `cudaFree(nullptr)`) well before we call this API function, we should be ok right? Since this file is CUDA-only I don't think it's a problem to directly call `cuProfilerStart`, but I think it'd be better to not have to link in another library...

Reply:
Yeah it's fine, actually,  both `cu(da)ProfilerStart` have this doc: 

> Enables profile collection by the active profiling tool for the current context.

So it's not even clear if `cuProfilerStart` would work without a context.